daemon/vpn/nordlynx/libtelio/libtelio.go

```diff
@@ -247,24 +250,24 @@ func (l *Libtelio) Start(
 		return fmt.Errorf("opening the tunnel: %w", err)
 	}
 
+	l.currentServer = serverData
 	if err = l.connect(serverData.IP, serverData.NordLynxPublicKey); err != nil {
 		return err
 	}
 
 	// Remember the values used for connection. This is necessary
 	// in case meshnet is enabled and disabled before calling Stop
 	l.currentPrivateKey = creds.NordLynxPrivateKey
-	l.currentServerIP = serverData.IP
-	l.currentServerPublicKey = serverData.NordLynxPublicKey
 	return nil
 }
 
 // connect to the VPN server
 func (l *Libtelio) connect(serverIP netip.Addr, serverPublicKey string) error {
 	ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
-	defer cancel()
+	l.cancelConnectionMonitor = cancel
```

Comment:
shouldn't this be moved after the if from 272 `if err := toError(l.lib.ConnectToExitNode(`? otherwise in case of an error it might generate a `disconnected` at `cancel()`

Reply:
I think disconnect could be valid in some cases(for example when connecting event has already been sent). I have added state tracking in the connection monitor function, so now disconnect will be sent only when previous published state was connected or connecting.